pkg/plugin/manager/manager.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 Comcast Cable Communications Management, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"fmt"
+	"sync"
+
+	goplugin "plugin"
+
+	"github.com/xmidt-org/ears/pkg/filter"
+	"github.com/xmidt-org/ears/pkg/plugin"
+	"github.com/xmidt-org/ears/pkg/receiver"
+	"github.com/xmidt-org/ears/pkg/sender"
+)
+
+var _ Manager = (*manager)(nil)
+
+type manager struct {
+	sync.Mutex
+	pathPrefix string
+
+	// Map of the name used in the configuration to the loaded plugin
+	registrations map[string]Registration
+}
+
+type Option func(OptionProcessor) error
+type OptionProcessor interface{}
+
+func NewManager(options ...Option) *manager {
```

Comment:
Who is calling the NewManager? It wasn't obvious to me who instantiate the plugin manager.

Reply:
Also, if we want to incorporate this into uberfx, I don't think uberfx supports variadic arguments.

Reply:
Correct.  I'll find a project that talks about that -- That person's design is to create uberfx specific packages.  Those packages would leverage the object's proper interface.  That way, the project does not get polluted by uberfx code.  Let me find that guide.

Reply:
@holidaymike  I can't find the original blog I was reading, but the gist is to create an `uber.fx` specific package.

* https://blog.huyage.dev/posts/simple-dependency-injection-in-go/
* https://github.com/yagehu/sample-fx-app/blob/master/internal/loggerfx/module.go

The loggerfx package above is the general pattern.  You would not expect `zap` to support `uber.fx`  directly.  There should also be a translation of the CLI arguments to the instantiated object (via a `New` + `options` design).  I expect something like `cmd/ears/uberfx/*.go` to be a collection of code that assists in this process.

Reply:
@holidaymike And actually, you have already implemented that pattern.  https://github.com/xmidt-org/ears/blob/plugin-manager/internal/pkg/app/app.go

Reply:
https://github.com/preslavmihaylov/fxappexample